test/generaloption.py

```diff
@@ -855,11 +869,15 @@ def test_option_as_value(self):
         self._match_testoption1_sysexit(['--store', '-b'],
                                         "Value '-b' is also a valid option")
 
+        # same for --store=-b
+        self._match_testoption1_sysexit(['--store=-b'],
```

Comment:
yes, indeed, that's also why my modified stuff doesn't pass the tests, will push another commit to fix this